src/Common/src/Interop/Unix/System.Native/Interop.MessageHeader.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        internal unsafe struct IOVector
+        {
+            public byte* Base;
+            public UIntPtr Count;
```

Comment:
I was told by @nguerrera that we decided not to use IntPtr for size_t conversions. Here are the quotes from him:

>1. ... we pick a fixed size in the shim. Either 32 or 64 bit. If it's just a scratch buffer, etc. We use 32-bit.
2. However, if we really need 64-bit (memory mapped files, then we use that.
3. ...< snip >...  You can do that. There's no hard rule other than no variable size_t in shim sig, and no silent truncation.